services/edus/src/test/java/app/coronawarn/datadonation/services/edus/otp/OtpRedemptionIntegrationTest.java

```diff
@@ -66,6 +64,50 @@ void testShouldReturnResponseStatusCode200AndStateValidWhenNotRedeemed() throws
         .andExpect(MockMvcResultMatchers.jsonPath("$.state").value("valid"));
   }
 
+  @Test
+  void testStrongClientIntegrityCheckForAndroid() throws Exception {
+    OtpRedemptionRequest validOtpRedemptionRequest = new OtpRedemptionRequest();
+    validOtpRedemptionRequest.setOtp(VALID_UUID);
+
+    OneTimePassword otpWithValidAndroidStrongIntegrityCheck = createOtp(VALID_UUID, LocalDateTime.now().plusDays(5), null);
+    otpWithValidAndroidStrongIntegrityCheck.setAndroidPpacBasicIntegrity(true);
+    otpWithValidAndroidStrongIntegrityCheck.setAndroidPpacCtsProfileMatch(true);
+    otpWithValidAndroidStrongIntegrityCheck.setAndroidPpacEvaluationTypeHardwareBacked(true);
+
+    when(otpRepository.findById(any())).thenReturn(Optional.of(otpWithValidAndroidStrongIntegrityCheck));
+
+    mockMvc.perform(MockMvcRequestBuilders
+        .post(OTP_REDEEM_URL)
+        .content(asJsonString(validOtpRedemptionRequest))
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(MockMvcResultMatchers.jsonPath("$.state").value("valid"))
+        .andExpect(MockMvcResultMatchers.jsonPath("$.strongClientIntegrityCheck").value(true));
+  }
+
+  @Test
+  void testInvalidStrongClientIntegrityCheckForAndroid() throws Exception {
```

Comment:
@mibrasap test case for iOS would be nice 👍

Reply:
Done